tests-functional/conftest.py

```diff
@@ -2,6 +2,49 @@
 from dataclasses import dataclass, field
 from typing import List
 import pytest
+import logging
+
+
+# Apply the filter immediately to catch all logs
+class ConnectionNoiseFilter(logging.Filter):
+    def filter(self, record):
+        msg = str(record.msg)
+
+        # Suppress connection reset errors during teardown
+        if "Connection reset by peer" in msg:
+            return False
+        # Suppress websocket connected warnings
+        if "websocket connected" in msg.lower():
+            return False
+        # Suppress all connection lost messages during teardown
+        if "Connection to remote host was lost" in msg:
+            return False
+        # Suppress connection aborted errors
+        if "Connection aborted" in msg:
+            return False
+        # Suppress any websocket-related errors during teardown
+        if "websocket" in msg.lower() and record.levelno >= logging.WARNING:
+            return False
+        # Suppress any connection-related errors during teardown
+        if any(term in msg.lower() for term in ["connection", "remote host", "goodbye"]) and record.levelno >= logging.WARNING:
+            return False
+        return True
+
+
+# Apply the filter to root logger immediately
+logging.getLogger().addFilter(ConnectionNoiseFilter())
+
+# Suppress problematic setup/teardown logs
+w = logging.getLogger("websocket")
+w.addFilter(ConnectionNoiseFilter())
+
+logging.getLogger("urllib3.connectionpool").setLevel(logging.ERROR)
+logging.getLogger("urllib3").setLevel(logging.ERROR)
+logging.getLogger("requests.packages.urllib3").setLevel(logging.ERROR)
+logging.getLogger("requests").setLevel(logging.WARNING)
+
+# Suppress any other connection-related warnings
+logging.getLogger("websocket-client").setLevel(logging.ERROR)
```

Comment:
I think this should be enough to filter all other things like `websocket connected` messages, no? 🤔